### PR TITLE
Update and export some props and convert ProgressBar to TypeScript

### PR DESCRIPTION
**Pull request by pvenable**
The need for these changes arose during the new home page development in the main app.

### CODE REVIEW

File: dist/index.d.ts [hunk 1]
```diff
@@ -4,9 +4,9 @@ export * from './Alert';
 export { default as AutoComplete, StyledAutoComplete, styledAutoCompleteComponents } from './AutoComplete';
 export { default as Avatar } from './Avatar';
 export { default as Badge } from './Badge';
-export { default as Box } from './Box';
+export { default as Box, BoxProps } from './Box';
 export { default as Button, ButtonProps, ButtonSize, ButtonVariant } from './Button';
-export { default as Card } from './Card';
+export { default as Card, CardProps } from './Card';
```
Comment (pvenable) [inline dist/index.d.ts:9]: Export `BoxProps` and `CardProps`.

File: src/Heading/Heading.tsx [hunk 2]
```diff
@@ -26,13 +27,15 @@ const calculatedMargin = ({
...
+export type Props = TextProps & {
   fontSize: FontSize;
   fontWeight?: FontWeight;
   mb?: string;
 };
 
-const Heading = styled(Text)<Props>`
+const TextWithColorPropFix: WithColorPropFix<typeof Text> = Text;
+
+const Heading = styled(TextWithColorPropFix)<Props>`
```
Comment (pvenable) [inline src/Heading/Heading.tsx:38]: These changes were made to support the `color` prop on e.g. `Heading.H2`.

File: src/Heading/__tests__/Heading.test.tsx [hunk 3]
```diff
@@ -5,7 +5,7 @@ import Heading from '../Heading';
 (['H1', 'H2', 'H3', 'H4', 'H5', 'H6'] as const).forEach((variant) => {
   test(`renders a ${variant}`, () => {
     const Variant = Heading[variant];
-    const tree = createWithTheme(<Variant>Hello World</Variant>);
+    const tree = createWithTheme(<Variant color="white">Hello World</Variant>);
```
Comment (pvenable) [inline src/Heading/__tests__/Heading.test.tsx:8]: Updated the test to exercise the `color` prop.  This wouldn't have compiled before the change.

File: src/ProgressBar/ProgressBar.tsx [hunk 4]
```diff
@@ -26,13 +27,24 @@ const BarForeground = styled(Box)`
   border-radius: inherit;
 `;
 
-const percentageText = (percentage) => (
-  <Text flex="0" ml="smaller" fontSize="size3" fontWeight="medium">
```
Comment (pvenable) [inline src/ProgressBar/ProgressBar.tsx:30]: `Text` doesn't support `flex`; this doesn't appear to hurt anything.

File: src/ProgressBar/ProgressBar.tsx [hunk 5]
```diff
@@ -45,7 +57,6 @@ const ProgressBar = ({ percentage, iconEndName, showPercentage, ...props }) => (
     {iconEndName && (
       <Icon
         ml="smaller"
-        flex="0"
```
Comment (pvenable) [inline src/ProgressBar/ProgressBar.tsx:48]: `Icon` also doesn't support `flex`, and this also doesn't appear to hurt anything.

File: src/index.ts [hunk 6]
```diff
@@ -8,14 +8,14 @@ export {
...
-export { default as Box } from './Box';
+export { default as Box, BoxProps } from './Box';
 export {
   default as Button,
   ButtonProps,
   ButtonSize,
   ButtonVariant
 } from './Button';
-export { default as Card } from './Card';
+export { default as Card, CardProps } from './Card';
```
Comment (pvenable) [inline src/index.ts:18]: Export `BoxProps` and `CardProps`.

**Review by xxx — APPROVED** [review]

**Pull request merged**